src/services/session.service.js

```diff
@@ -1,34 +1,163 @@
-const CorbadoApi = require('./CorbadoApi');
+const assert = require("assert");
+const jose = require("jose");
+const User = require("../entities/User");
 
 class SessionService {
-    constructor(projectID, apiSecret, apiURL) {
-        this.corbadoApi = new CorbadoApi(projectID, apiSecret,apiURL);
+
+    #version;
+    #shortSessionCookieName;
+    #issuer;
+    #jwksURI;
+    #cacheMaxAge;
+    #client;
+
+    /**
+     * Create a session service.
+     * @param version
+     * @param shortSessionCookieName
+     * @param issuer
+     * @param jwksURI
+     * @param cacheMaxAge
+     * @param client
+     */
+    constructor(version, shortSessionCookieName, issuer, jwksURI, cacheMaxAge, client) {
+        if (!version || !client) {
+            throw new Error('Invalid argument(s)');
+        }
+        this.#version = version;
+        this.#shortSessionCookieName = shortSessionCookieName;
+        this.#issuer = issuer;
+        this.#jwksURI = jwksURI;
+        this.#cacheMaxAge = cacheMaxAge
+        this.#client = client;
+    }
+
+    /**
+     * Obtain the short session cookie.
+     * @param req
+     * @returns {*|string|null}
+     */
+    getShortSessionValue(req) {
+        if (this.#version === 'v1') {
```

Comment:
i would say we implement 2 classes in order to abstract this better.

Reply:
we change the validation part only atm

Reply:
ok